includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -527,13 +534,11 @@ static function ( $validation_error ) {
 					);
 				} else {
 					// Otherwise, we have a rare condition where multiple tag specs fail for different reasons.
-					$this->remove_invalid_child(
-						$node,
-						[
-							'code'   => self::DISALLOWED_TAG_MULTIPLE_CHOICES,
-							'errors' => $validation_errors,
-						]
-					);
+					foreach ( $validation_errors as $validation_error ) {
+						if ( true === $this->remove_invalid_child( $node, $validation_error ) ) {
+							break; // Once removed, ignore remaining errors.
+						}
+					}
```

Comment:
Given a Custom HTML block with the following markup:

```html
<amp-accordion id="my-accordion" disable-session-states="">
  <section>
    <h2>Section 1</h2>
    <p>Content in section 12.</p>
  </section>
  <section>
    <h2>Section 2</h2>
    <div>Content in section 2.</div>
  </section>
  <section expanded="">
    <!--<h2>Section 3</h2>-->
    <amp-img src="/static/inline-examples/images/squirrel.jpg" width="320" height="256"></amp-img>
  </section>
</amp-accordion>
```

This is invalid because of the `h2` being commented out. The `amp-accordion` component requires an `h2` followed by a `div` or a `p`. The validation errors for this are not expected, however:

![image](https://user-images.githubusercontent.com/134745/77219510-c1d85980-6af3-11ea-9299-2b29cf89ba11.png)

Notice the `amp-img` lacks the expected Custom HTML source.

I've fixed this here in feababc by breaking the loop of validation error reporting once a markup causing validation error is removed.

This will probably make this code no longer necessary:

https://github.com/ampproject/amp-wp/blob/3c29860c139d0706f9bb1029df0752803e625681/includes/sanitizers/class-amp-base-sanitizer.php#L477-L481

Though it doesn't hurt to have it in there.

Reply:
> Notice the `amp-img` lacks the expected Custom HTML source.

Good catch!

> This will probably make this code no longer necessary:

In this context yes, but it could potentially prevent an error being thrown when attempting to remove a node multiple times.